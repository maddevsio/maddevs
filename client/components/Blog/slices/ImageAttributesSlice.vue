<template>
  <div>
    <template v-if="$prismic.asText(url)">
      <div class='post-part single'>
        <a class="block-img" :href="$prismic.asText(url)" :target="target">
          <prismic-image :field="img" :title="$prismic.asText(title)"/>
        </a>
      </div>
    </template>
    <template v-else>
      <div class='post-part single'>
        <p class="block-img">
          <prismic-image :field="img" :title="$prismic.asText(title)"/>
        </p>
      </div>
    </template>
  </div>
</template>

<script>
export default {
  name: 'image-attributes-slice',
  props: ['slice'],
  data: function() {
    return {
      img: '',
      title: '',
      url: '',
      target: ''
    };
  },
  created() {
    this.img = this.slice.primary.image;
    this.title = this.slice.primary.title;
    this.url = this.slice.primary.url;
    this.target = this.slice.primary.target;
  }
};
</script>

<style scoped lang="scss">
  /deep/ img {
    max-width: 100%;
    height: auto;
  }
</style>
