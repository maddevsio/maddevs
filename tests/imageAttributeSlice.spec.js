import {
  mount
} from '@vue/test-utils';
import ImageAttributesSlice from '../client/components/Blog/slices/ImageAttributesSlice';

describe('Image attribute slice component', () => {
  let wrapper;

  const slice = {
    items: [{}],
    primary: {
      image: {
        alt: 'alt text',
        copyright: null,
        dimensions: {height:298, width:1200},
        url: 'https://images.prismic.io/superpupertest/4b27a325-c48e-4f20-943d-e40537291055_2020-12-08_14-11.png?auto=compress,format&rect=0,0,1231,306&w=1200&h=298'
      },
      target: '_blank',
      title: [{
        spans: [],
        text: 'Tired banner',
        type: 'paragraph'
      }],
      url: 'https://blog.maddevs.io'
    },
    slice_label: null,
    slice_type: 'image_with_attributes'
  };

  beforeEach(() => {
    wrapper = mount(ImageAttributesSlice, {
      propsData: {slice},
      mocks: {
        $prismic: {
          asText: text => text
        }
      },
      stubs: ['prismic-image']
    });
  });

  test('is vue component', () => {
    expect(wrapper.exists()).toBeTruthy();
  });

  test('renders correctly', () => {
    expect(wrapper.element).toMatchSnapshot();
  });

  test('link has target attribute', () => {
    expect(wrapper.find('.block-img').attributes('target')).toMatch('_blank');
  });

  test('link has href attribute', () => {
    expect(wrapper.find('.block-img').attributes('href')).toMatch('https://blog.maddevs.io');
  });
});
