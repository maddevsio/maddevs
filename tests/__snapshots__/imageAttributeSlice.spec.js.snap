// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Image attribute slice component renders correctly 1`] = `
<div>
  <div
    class="post-part single"
  >
    <a
      class="block-img"
      href="https://blog.maddevs.io"
      target="_blank"
    >
      <prismic-image-stub
        field="[object Object]"
        title="[object Object]"
      />
    </a>
  </div>
</div>
`;
